mesonbuild/interpreter/interpreter.py

```diff
@@ -3086,8 +3085,8 @@ def build_both_libraries(self, node, args, kwargs):
             static_lib.sources = []
             static_lib.generated = []
             # Compilers with no corresponding sources confuses the backend.
-            # Keep only the first compiler because it is the linker.
-            static_lib.compilers = dict(itertools.islice(static_lib.compilers.items(), 1))
+            # Keep only compilers used for linking
+            static_lib.compilers = {k: v for k, v in static_lib.compilers.items() if k in compilers.clink_langs}
```

Comment:
Hmm, what was the original issue with confusing the backend anyway?

Reply:
ninja backend does a bunch of `if 'vala' in target.compilers` etc... But maybe instead of keeping the first compiler I think I should keep only the last actually.

Reply:
Aha, makes sense.

In that case sticking to just the ones that are valid linker drivers seems like an equivalent good solution.